src/OrchardCore.Modules/OrchardCore.Media/Controllers/AdminController.cs

```diff
@@ -140,75 +143,79 @@ public async Task<ActionResult<object>> GetMediaItem(string path)
 
         [HttpPost]
         [MediaSizeLimit]
-        public async Task<ActionResult<object>> Upload(
-            string path,
-            string contentType,
-            ICollection<IFormFile> files)
+        public async Task<IActionResult> Upload(string path)
         {
             if (!await _authorizationService.AuthorizeAsync(User, Permissions.ManageMedia))
             {
                 return Forbid();
             }
 
-            if (string.IsNullOrEmpty(path))
-            {
-                path = "";
-            }
-
-            var result = new List<object>();
-
-            // Loop through each file in the request
-            foreach (var file in files)
```

Comment:
If the XXXMediaProvider already supports upload as chunks we could reuse the old code because the underlying provider will take care of chunking otherwise, we need to implement it ourselves (in the case of FileSystem)

Reply:
See my comments above https://github.com/OrchardCMS/OrchardCore/pull/13903#discussion_r1254726596